src/libraries/sendtohelixhelp.proj

```diff
@@ -144,18 +144,18 @@
     <DotNetCliPackageType>sdk</DotNetCliPackageType>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(UseDotNetCliVersionFromGlobalJson)' == 'true'">
-    <GlobalJsonContent>$([System.IO.File]::ReadAllText('$(RepoRoot)global.json'))</GlobalJsonContent>
-    <DotNetCliVersion>$([System.Text.RegularExpressions.Regex]::Match($(GlobalJsonContent), '(%3F&lt;="dotnet": ").*(%3F=")'))</DotNetCliVersion>
-  </PropertyGroup>
-
   <ItemGroup>
     <HelixProperties Condition="'$(RuntimeFlavor)' != ''" Include="runtimeFlavor" Value="$(RuntimeFlavor)" />
     <HelixProperties Condition="'$(Scenario)' != ''" Include="scenario" Value="$(Scenario)" />
   </ItemGroup>
 
   <!-- Ensure that all HelixPreCommand items are ready before this -->
   <Target Name="BuildHelixCommand">
+    <!-- setting DotNetCliVersion here to ensure that it is set to the intended value -->
+    <PropertyGroup Condition="'$(UseDotNetCliVersionFromGlobalJson)' == 'true'">
+      <GlobalJsonContent>$([System.IO.File]::ReadAllText('$(RepoRoot)global.json'))</GlobalJsonContent>
+      <DotNetCliVersion>$([System.Text.RegularExpressions.Regex]::Match($(GlobalJsonContent), '(%3F&lt;="dotnet": ").*(%3F=")'))</DotNetCliVersion>
+    </PropertyGroup>
```

Comment:
@premun @radical with #68024 merged, I think we can revert this right?

Reply:
I think we need to rebase this and move onto .NET 7.0 XHarness in this PR again?

Reply:
Good idea. Still we need to clean up the TODOs a bit. I can do that later in this PR

Reply:
I would remove them as there are other settings that need to be brought back as well when we transition from a major version to another (ie the installer tests require some hand work). Usually when I do such a change I just look into git history and find the relevant things to bring back.

Reply:
Sure, makes sense. I agree that the project files will be probably much different 1 year from now so looking up the commit is probably more meaningful.

I have reverted the changes to the state of main before we started all this.

I renamed the PR so it's easier to find 1 year from now

Reply:
To correct myself - I reverted only changes from #68024 (not the ones made in this PR)